src/OmniSharp.Abstractions/Models/v1/AutoComplete/AutoCompleteResponse.cs

```diff
@@ -24,7 +24,8 @@ public class AutoCompleteResponse
         public override bool Equals(object other)
         {
             var otherResponse = other as AutoCompleteResponse;
-            return otherResponse.DisplayText == DisplayText
+            return otherResponse != null
+                && otherResponse.DisplayText == DisplayText
```

Comment:
I believe this is the only actual change in this PR. Please revert all the other changes: I'd be fine with them in a separate PR, but the large number of unrelated changes makes it more difficult to understand what the bug is and what changed here. I'd also like to see a test showing what bugs this was causing.

Reply:
I am not sure whether this deserve a test. I did not meet any real world case and just did the changes by the way when reading code.
Anyway, this shows the difference:
```C#
    public unsafe static void Main()
    {
        AutoCompleteResponse foo1 = new();
        Object foo2 = new();
        TestEqual(foo1, foo2); // before the change, console logs "error", for we try to access a property of null.
                               // after the change, console logs "not equal"
    }

    public static void TestEqual(Object o1, Object o2)
    {
        try
        {
            if (!o1.Equals(o2))
            {
                Console.WriteLine("not equal");
            }
        }
        catch
        {
            Console.WriteLine("error");
        }
    }
```

Reply:
Alright. Please revert the rest of the changes though.